### PR TITLE
When building a tarball, avoid trying to copy submodules

### PR DESCRIPTION
**Pull request by levitte**
submodules are directories that we don't want in our tarballs, so
avoid them.

-----

This is an alternative to #4114

### CODE REVIEW

File: Configurations/unix-Makefile.tmpl [hunk 1]
```diff
@@ -686,8 +686,10 @@ tar:
 	DISTDIR=$(NAME); \
 	mkdir -p $$TMPDIR/$$DISTDIR; \
 	(cd $(SRCDIR); \
+	 submodule_re=`git submodule status | sed -e 's/^.//' | cut -d' ' -f2`; \
+	 submodule_re="^(`echo $$submodule_re | sed -e 's/ /|/g'`)\$$"; \
 	 git ls-tree -r --name-only --full-tree HEAD \
-         | grep -v '^fuzz/corpora' \
+	 | grep -v '^fuzz/corpora' | egrep -v "$$submodule_re" \
```
Comment (dot-asm) [inline Configurations/unix-Makefile.tmpl]: Does everybody have egrep? But in either case why grep in one case and egrep in another? And why not simply incorporate fuzz/corpora into $$submodule_re?

Comment (dot-asm) [inline Configurations/unix-Makefile.tmpl]: Majority seems to have egrep, but I'm not convinced that all greps accept regular expression. In other words suggestion is kind of change to single egrep.

Comment (dot-asm) [inline Configurations/unix-Makefile.tmpl]: Oh! Actually manual pages call egrep obsolescent and deprecated. But they say that if you have expression, then you should denote it with -E. By this logic egrep should go, but you should use -E. Leaning something new every day...

Comment (dot-asm) [inline Configurations/unix-Makefile.tmpl]: > if you have expression

Extended expression like one in question.

Comment (levitte) [inline Configurations/unix-Makefile.tmpl]: The submodule regexp is one ending in a `$`, while the fuzz corpora one isn't.  Of course, I could make the regexp end up like this:

```
^(boringssl$|krb5$|pyca-cryptography$|fuzz/corpora)
```

but that makes for a slightly more complex generation of the regexp, and I didn't find that worth the while.

Regard egrep vs -E, how far back did -E exist?  Will it work on all platforms we support?

Comment (levitte) [inline Configurations/unix-Makefile.tmpl]: Another look and I realised the generation of the regexp didn't become as complex as I thought...  hold on.

Comment (levitte) [inline Configurations/unix-Makefile.tmpl]: Okie, fix coming up

Comment (dot-asm) [inline Configurations/unix-Makefile.tmpl]: > The submodule regexp is one ending in a `$`

But it doesn't have to, `^` leaves no ambiguity. As for -E, Tru64, HP-UX, AIX, Solaris 10, MacOS X, Linux from before 2000...

**Comment by dot-asm** [comment]
This looks proper, but there was another problem in #4114, namely Solaris doesn't know --owner flag...

**Comment by levitte** [comment]
That's a different problem, which I'm not addressing here.

**Comment by dot-asm** [comment]
> Solaris doesn't know --owner flag

As a matter of fact presumably it's GNU thing, so Solaris is probably not the only one that doesn't...

**Review by dot-asm — APPROVED** [review]

**Comment by levitte** [comment]
Merged.  5b7b0115256c4d0be0f92468df14bdf1c57634a5

File: Configurations/unix-Makefile.tmpl (hunk 1 above)
Comment (richsalz) [inline Configurations/unix-Makefile.tmpl]: "sed -e /xxxx/d -e /yyyy/d ..."

**Pull request closed without merging**